kudu/pom.xml

```diff
@@ -0,0 +1,164 @@
+    <version>3.8.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>malhar-kudu</artifactId>
+  <name>Apache Apex Malhar Kudu Support</name>
+  <packaging>jar</packaging>
+
+  <properties>
+    <checkstyle.console>false</checkstyle.console>
+  </properties>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.antlr</groupId>
+        <artifactId>antlr4-maven-plugin</artifactId>
+        <version>4.7</version>
+        <executions>
+          <execution>
+            <id>kudusqlantlr4parsergen</id>
+            <goals>
+              <goal>antlr4</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <configuration>
+          <logViolationsToConsole>${checkstyle.console}</logViolationsToConsole>
+          <excludes>**/KuduSQLExpression*</excludes> <!--To account for violations from the Antlr autogenerated code -->
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.rat</groupId>
+        <artifactId>apache-rat-plugin</artifactId>
+        <configuration>
+          <excludes combine.children="append">
+            <exclude>src/main/java/org/apache/apex/malhar/kudu/sqlparser/KuduSQLExpression.tokens</exclude>
+            <exclude>src/main/java/org/apache/apex/malhar/kudu/sqlparser/KuduSQLExpressionLexer.tokens</exclude>
+            <exclude>src/main/java/org/apache/apex/malhar/kudu/sqlparser/KuduSQLExpressionLexer.java</exclude>
+            <exclude>src/main/java/org/apache/apex/malhar/kudu/sqlparser/KuduSQLExpressionListener.java</exclude>
+            <exclude>src/main/java/org/apache/apex/malhar/kudu/sqlparser/KuduSQLExpressionParser.java</exclude>
+            <exclude>src/main/antlr4/KuduSQLExpression.g4</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>add-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${basedir}/target/generated-sources/antlr4</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
+  <dependencies>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>malhar-library</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>malhar-library</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+      <classifier>tests</classifier>
+    </dependency>
+    <dependency>
```

Comment:
Shouldn't the scope be provided

Reply:
No, it is a test dependency for testing only. Besides, malhar dependencies are never provided scope.

Reply:
Not malhar, referring to apex-common

Reply:
Yep, and that one is another potential candidate for parent pom, in case it is used in most modules (to be investigated).

Reply:
On a tangential note, I don't see much benefit of separating api and common anymore as practically all apps use common as a dependency instead of api, and I don't know of an instance where api is used standalone. Common can be merged into api and an empty artifact that depends on the api, can be left in place for backward compatibility